src/Grids/orthogonal_spherical_shell_grid.jl

```diff
@@ -20,27 +20,27 @@ struct OrthogonalSphericalShellGrid{FT, TX, TY, TZ, Z, A, C, Arch} <: AHCG{FT, T
        Hy :: Int
        Hz :: Int
        Lz :: FT
-     λᶜᶜᵃ :: A
-     λᶠᶜᵃ :: A
-     λᶜᶠᵃ :: A
-     λᶠᶠᵃ :: A
-     φᶜᶜᵃ :: A
-     φᶠᶜᵃ :: A
-     φᶜᶠᵃ :: A
-     φᶠᶠᵃ :: A
+     λᶜᶜᵃ :: CC
+     λᶠᶜᵃ :: FC
+     λᶜᶠᵃ :: CF
+     λᶠᶠᵃ :: FF
+     φᶜᶜᵃ :: CC
+     φᶠᶜᵃ :: FC
+     φᶜᶠᵃ :: CF
+     φᶠᶠᵃ :: FF
         z :: Z
-    Δxᶜᶜᵃ :: A
-    Δxᶠᶜᵃ :: A
-    Δxᶜᶠᵃ :: A
-    Δxᶠᶠᵃ :: A
-    Δyᶜᶜᵃ :: A
-    Δyᶜᶠᵃ :: A
-    Δyᶠᶜᵃ :: A
-    Δyᶠᶠᵃ :: A
-    Azᶜᶜᵃ :: A
-    Azᶠᶜᵃ :: A
-    Azᶜᶠᵃ :: A
-    Azᶠᶠᵃ :: A
+    Δxᶜᶜᵃ :: CC
+    Δxᶠᶜᵃ :: FC
+    Δxᶜᶠᵃ :: CF
+    Δxᶠᶠᵃ :: FF
+    Δyᶜᶜᵃ :: CC
+    Δyᶠᶜᵃ :: FC
+    Δyᶜᶠᵃ :: CF
```

Comment:
@simone-silvestri @navidcy did you two notice that the ordering was reversed for these two metrics in the definition of OrthogonalSphericalShellGrid? This was not a bug but rather simply a glitch in the pattern. It seems like a major source of bugs though, so I fixed it.

Reply:
Hm... Don't remember if I had a reason doing it like that or it was a glitch. You can see this pattern elsewhere in the file, e.g.,

https://github.com/CliMA/Oceananigans.jl/blob/3c6f4212575e2da844b471ffd8bcb89979895ca7/src/Grids/orthogonal_spherical_shell_grid.jl#L225-L226

 so it might have something to do with symmetry or similarity that x-fc and y-cf have (or the similarity they have in the cubed sphere grid). Either way, good that you smoothed it out! Let's fix it everywhere?

Reply:
Since the tests pass, I think all substantial places in the code are fixed. However there may be further cosmetic issues which I hope can be fixed in a subsequent PR.